fleetspeak/src/server/spanner/filestore.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"time"
+
+	"cloud.google.com/go/spanner"
+
+	"github.com/google/fleetspeak/fleetspeak/src/server/db"
+
+	tpb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// StoreFile implements db.FileStore.
+func (d *Datastore) StoreFile(ctx context.Context, service, name string, data io.Reader) error {
+	b, err := io.ReadAll(data)
+	if err != nil {
+		return err
+	}
+
+	_, err = d.dbClient.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		d.tryStoreFile(txn, service, name, b)
+		return nil
```

Comment:
```suggestion
		return d.tryStoreFile(txn, service, name, b)
```

`tryStoreFile` should return an error.